plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -1694,3 +1749,95 @@ def delete_component(self, cid, realm='master'):
+                            data=json.dumps(userrep), validate_certs=self.validate_certs)
+        except Exception as e:
+            self.module.fail_json(msg="Could not create user %s in realm %s: %s"
+                                      % (userrep['user_username'], realm, str(e)))
+
+    def get_users_by_username(self, username, realm='master'):
+        """ Fetch the name and ID of user by username on the Keycloak server.
+
+        To fetch the full data of the user.
+
+        :param username: The username of the user. A lookup will be performed to retrieve the user ID.
+        :param realm: Return the groups of this realm (default "master").
+        """
+        users_url = URL_USERS.format(
+            url=self.baseurl, realm=realm) + '?' + urlencode(dict(username=username))
+        try:
+            return json.loads(to_native(open_url(users_url, method="GET", headers=self.restheaders,
+                                                 validate_certs=self.validate_certs).read()))
+        except HTTPError as e:
+            if e.code == 404:
+                return None
+            else:
+                self.module.fail_json(msg='Could not fetch user %s in realm %s: %s'
+                                          % (username, realm, str(e)))
+        except Exception as e:
+            self.module.fail_json(msg='Could not fetch user %s in realm %s: %s'
+                                      % (username, realm, str(e)))
+
+    def update_user(self, userrep, realm="master"):
+        """ Update an existing user.
+
+        :param userrep: A UserRepresentation of the updated user.
+        :return HTTPResponse object on success
+        """
+        user_url = URL_USER.format(
+            url=self.baseurl, realm=realm, id=userrep['id'])
+
+        try:
+            return open_url(user_url, method='PUT', headers=self.restheaders,
+                            data=json.dumps(userrep), validate_certs=self.validate_certs)
+        except Exception as e:
+            self.module.fail_json(msg='Could not update user %s in realm %s: %s'
+                                      % (userrep['username'], realm, str(e)))
+
+    def delete_user(self, username=None, id=None, realm="master"):
+        """ Delete a user. One of username or id must be provided.
+
+        Providing the user ID is preferred as it avoids a second lookup to
+        convert a username to an ID.
+
+        :param username: The username of the user. A lookup will be performed to retrieve the user ID.
+        :param id: The ID of the user (preferred to username).
+        :param realm: The realm in which this user resides, default "master".
+        """
+
+        if id is None and username is None:
+            # prefer an exception since this is almost certainly a programming error in the module itself.
+            raise Exception(
+                "Unable to delete user - one of user ID or username must be provided.")
```

Comment:
This test for is or username being present should also exist for `update_user`.

We should also throw an error if both are provided.